File: cpp/logbrew-cpp/examples/readme_example.cpp
#include "logbrew.hpp"

#include <cstdlib>
#include <iostream>

int main() {
  try {
    logbrew::LogBrewClient client(logbrew::Config{"LOGBREW_API_KEY", "logbrew-cpp", logbrew::version, 2});

    client.release(
        "evt_release_001",
        "2026-06-02T10:00:00Z",
        logbrew::ReleaseAttributes{"1.2.3", "abc123def456", "Public release marker"});
    client.environment(
        "evt_environment_001",
        "2026-06-02T10:00:01Z",
        logbrew::EnvironmentAttributes{"production", "global"});
    client.issue(
        "evt_issue_001",
        "2026-06-02T10:00:02Z",
        logbrew::IssueAttributes{"Checkout timeout", "error", "Request timed out after retry budget"});
    client.log(
        "evt_log_001",
        "2026-06-02T10:00:03Z",
        logbrew::LogAttributes{"worker started", "info", "job-runner"});
    client.span(
        "evt_span_001",
        "2026-06-02T10:00:04Z",
        logbrew::SpanAttributes{"GET /health", "trace_001", "span_001", std::nullopt, "ok", 12.5});
    client.action(
        "evt_action_001",
        "2026-06-02T10:00:05Z",
        logbrew::ActionAttributes{"deploy", "success"});

    std::cout << client.preview_json() << '\n';
    logbrew::RecordingTransport transport;
    const logbrew::TransportResponse response = client.flush(transport);
    std::cerr << "{\"ok\":true,\"status\":" << response.status_code << ",\"attempts\":" << response.attempts
              << ",\"events\":" << transport.sent_bodies().size() << "}\n";
    return 0;
  } catch (const logbrew::SdkException &error) {
    std::cerr << error.code() << ": " << error.what() << '\n';
    return 1;
  }
}

File: unity/logbrew-unity/Samples~/ReadmeExample/ReadmeExample.cs
#nullable enable

using System;
using LogBrew.Unity;

public static class ReadmeExample
{
    public static void Main()
    {
        var client = LogBrewUnity.CreateClient("LOGBREW_API_KEY", "logbrew-unity");
        EnqueueCanonicalEvents(client);
        Console.WriteLine(client.PreviewJson());
        var response = client.Flush(RecordingTransport.AlwaysAccept());
        Console.Error.WriteLine("{\"ok\":true,\"status\":" + response.StatusCode + ",\"attempts\":" + response.Attempts + ",\"events\":6}");
    }

    public static void EnqueueCanonicalEvents(LogBrewClient client)
    {
        client.Release("evt_release_001", "2026-06-02T10:00:00Z", ReleaseAttributes.Create("1.2.3").WithCommit("abc123def456").WithNotes("Public release marker"));
        client.Environment("evt_environment_001", "2026-06-02T10:00:01Z", EnvironmentAttributes.Create("production").WithRegion("global"));
        client.Issue("evt_issue_001", "2026-06-02T10:00:02Z", IssueAttributes.Create("Checkout timeout", "error").WithMessage("Request timed out after retry budget"));
        client.Log("evt_log_001", "2026-06-02T10:00:03Z", LogAttributes.Create("worker started", "info").WithLogger("job-runner"));
        client.Span("evt_span_001", "2026-06-02T10:00:04Z", SpanAttributes.Create("GET /health", "trace_001", "span_001", "ok").WithDurationMs(12.5));
        client.Action("evt_action_001", "2026-06-02T10:00:05Z", ActionAttributes.Create("deploy", "success"));
    }
}
